src/main/java/gym/crm/repository/TraineeDAO.java

```diff
@@ -1,49 +1,103 @@
 package gym.crm.repository;
 
+import gym.crm.exception.CustomNotFoundException;
 import gym.crm.model.Trainee;
+
+import jakarta.persistence.EntityManager;
+import jakarta.persistence.NoResultException;
+import jakarta.persistence.PersistenceContext;
+import jakarta.persistence.TypedQuery;
+import jakarta.transaction.Transactional;
+import lombok.RequiredArgsConstructor;
 import org.springframework.stereotype.Repository;
 
-import java.util.HashMap;
 import java.util.List;
-import java.util.Map;
 
 @Repository
```

Comment:
https://www.baeldung.com/java-dao-vs-repository , I agree with that but I propose renaming it from DAO to repositiry , because annotation repository will match with the class name